daffodil-core/src/main/scala/org/apache/daffodil/dsom/RuntimePropertyMixins.scala

```diff
@@ -655,6 +656,18 @@ trait LayeringRuntimeValuedPropertiesMixin
       Nope
     }
   }
+  
+  final lazy val maybeLayerTransformArgsEv = {
+    if(optionLayerTransformArgsRaw.isDefined){
```

Comment:
It is ok if newly added properties have defaults. 
The whole layering proposal is still a proposal. It's unclear if it will evolve to support a <dfdl:defineLayer name="..."> type annotation to group things. 
So I'm not worried about this sort of thing currently.